### PR TITLE
Restrict auth method listing for u_anon

### PR DESCRIPTION
**Pull request by jefferai**
This keeps only essential information in list output for auth methods
when the user is u_anon. Boundary's default is to allow listing of
auth methods for anonymous users to allow discovery of authentication
points, but there's no need to return more information about internal
configuration than necessary.

### CODE REVIEW

File: internal/servers/controller/handlers/options.go [hunk 1]
```diff
@@ -41,3 +42,11 @@ func WithLogger(logger hclog.Logger) Option {
 		o.WithLogger = logger
 	}
 }
+
+// WithAnonymousListing provides an option when creating responses to only include those
+// desired for listing to anonymous users.
+func WithAnonymousListing(anonListing bool) Option {
```
Comment (jimlambrt) [inline internal/servers/controller/handlers/options.go:48]: mind adding a test?

**Review by talanknight — DISMISSED** [review]

File: internal/servers/controller/handlers/options.go (hunk 1 above)
Comment (jimlambrt) [reply internal/servers/controller/handlers/options.go:48]: Not sure it needs to be exported.

**Review by jimlambrt — APPROVED** [review]
minor bits.

File: internal/servers/controller/handlers/options.go (hunk 1 above)
Comment (jefferai) [reply internal/servers/controller/handlers/options.go:48]: It does, because this is in `handlers`, and it's being used in actual handler implementations.

Comment (jefferai) [reply internal/servers/controller/handlers/options.go:48]: Test added.

**Pull request merged**